spec/models/user_option_spec.rb

```diff
@@ -18,6 +18,28 @@
 
   end
 
+  describe ".mailing_list_mode" do
+    let!(:forum_user) { Fabricate(:user) }
+    let!(:mailing_list_user) { Fabricate(:user) }
+
+    before do
+      forum_user.user_option.update(mailing_list_mode: false)
+      mailing_list_user.user_option.update(mailing_list_mode: true)
+    end
+
+    it "should return false when `SiteSetting.disable_mailing_list_mode` is enabled" do
+      SiteSetting.expects(:disable_mailing_list_mode).twice.returns(true)
```

Comment:
Yeah, we usually try to avoid mocking when it's not needed ;) 

Just use `SiteSetting.disable_mailing_list_mode = true` is enough.